src/Symfony/Component/DomCrawler/Form.php

```diff
@@ -333,8 +333,8 @@ public function offsetGet($name)
     /**
      * Sets the value of a field.
      *
-     * @param string       $name  The field name
-     * @param string|array $value The value of the field
+     * @param string $name  The field name
+     * @param mixed  $value The value of the field
```

Comment:
That's wrong. For Form fields, the value is always a `string` or an array of `string`. The current phpdoc is good.

Reply:
Hello fabpot. Thanks for answering back.. but it's hard for me to follow exactly your thought here..

First of all this is the code inside `offsetSet` 

```
$this->fields->set($name, $value);
```

The `set` function of fields function accepts a mixed argument as value.


Secondly can you check the related issue on PHPStan?

As I posted there I have a test that it is doing this:

```
        $viewCrawler = $this->client->request('GET', '/mypath');
        $documentUploadForm = $viewCrawler->selectButton('Upload your Documents')->form();

        $documentUploadForm['document_upload_form[proofOfIdentity]'] = new UploadedFile(__DIR__ . '/Fixtures/validPassport.png', 'passport.png', 'image/png');
        $documentUploadForm['document_upload_form[proofOfResidency]'] = new UploadedFile(__DIR__ . '/Fixtures/validElectricityBill.png', 'electricalBill.png', 'image/png');

        $this->client->getContainer()->get('guzzle.mock_handler')->append(new Response());

        $crawler = $this->client->submit($documentUploadForm);
```

If my PR is wrong can you explain to me how my test should be refactored to be compatible with the docblocks?

Reply:
I haven't tested it, but your code looks wrong to me. DomCrawler does not support UploadedFile instances. It probably works "by chance" as it implements `__toString()`:

```php
$documentUploadForm['document_upload_form[proofOfIdentity]'] = __DIR__ . '/Fixtures/validPassport.png';
$documentUploadForm['document_upload_form[proofOfResidency]'] = __DIR__ . '/Fixtures/validElectricityBill.png'
```